### DISCUSSION TITLE
Any way to check if BIOS or EFI from iPXE shell?

### DISCUSSION THREAD

**Discussion by J-A-M-E-5** [ipxe/ipxe#1029] [Q&A] [answered]
Other than embedding a script with a flag set to a different value in the EFI and BIOS based builds, I want to ask if there is a native already implemented way of checking if running on a EFI system (booted via EFI) or on a BIOS system, *before* DHCP request.

Something like what cpuid does in a script at runtime to check if CPU is i386 or x64, but to check if running from EFI or BIOS. I know DHCP server will identify this with client-arch option being 6,7 or 9 for EFI as it is sent as part of the request for a lease, however I was looking to find it out before even using the interfaces, let alone DHCP.

Thanks

**Comment by NiKiZe** [comment, marked as answer]
Sure, see https://ipxe.org/cfg/platform
Also try out the config command .

**Reply by J-A-M-E-5** [reply]
Thanks, and to think I had studied all the manual!
